04.ls/ls.rb

```diff
@@ -55,71 +65,75 @@ def check_stickybit_sgid_suid(oct_mode, permission)
-    File.lstat(file).nlink.to_s.rjust(max + 1)
-  end
+def get_link_number(file)
+  File.lstat(file).nlink.to_s
 end
 
-def get_user_name(files)
-  max = files.map { |file| Etc.getpwuid(File.lstat(file).uid).name }.max_by(&:length).length
-  files.map do |file|
-    Etc.getpwuid(File.lstat(file).uid).name.ljust(max + 1)
-  end
+def get_user_name(file)
+  Etc.getpwuid(File.lstat(file).uid).name
 end
 
-def get_group_name(files)
-  max = files.map { |file| Etc.getgrgid(File.lstat(file).gid).name }.max_by(&:length).length
-  files.map do |file|
-    Etc.getgrgid(File.lstat(file).gid).name.ljust(max + 1)
-  end
+def get_group_name(file)
+  Etc.getgrgid(File.lstat(file).gid).name
 end
 
-def get_byte(files)
-  max = files.map { |file| File.lstat(file).size.to_s }.max_by(&:length).length
-  files.map do |file|
-    File.lstat(file).size.to_s.rjust(max)
-  end
+def get_byte(file)
+  File.lstat(file).size.to_s
 end
 
-def get_last_modified_date(files)
-  files.map do |file|
-    modified_time = File.lstat(file).mtime
-    modified_time.to_date.between?(Date.today.prev_month(6), Date.today) ? modified_time.strftime('%b %e %H:%M') : modified_time.strftime('%b %e  %Y')
-  end
+def get_link_number_width(files)
+  files.map { |file| get_link_number(file) }.max_by(&:length).length
 end
 
-def get_file_name(files)
-  files.map do |file|
-    File.lstat(file).symlink? ? "#{file} -> #{File.readlink(file)}" : file
-  end
+def get_user_name_width(files)
+  files.map { |file| get_user_name(file) }.max_by(&:length).length
 end
 
-def main(column_number)
-  options = select_options
-  files = acquire_files(a_option: options[:a], r_option: options[:r])
-  if options[:l]
-    [get_file_mode(files), get_link_number(files),
-     get_user_name(files), get_group_name(files),
-     get_byte(files), get_last_modified_date(files),
-     get_file_name(files)].transpose.map { |file| file.join(' ') }.unshift("total #{files.inject(0) { |i, file| i + File.lstat(file).blocks }}")
-  else
-    generate_files_for_display(files, column_number)
+def get_group_name_width(files)
+  files.map { |file| get_group_name(file) }.max_by(&:length).length
+end
+
+def get_byte_width(files)
+  files.map { |file| get_byte(file) }.max_by(&:length).length
+end
+
+def get_last_modified_date(file)
+  modified_time = File.lstat(file).mtime
+  modified_time.to_date.between?(Date.today.prev_month(6), Date.today) ? modified_time.strftime('%b %e %H:%M') : modified_time.strftime('%b %e  %Y')
+end
+
+def get_file_name(file)
+  File.lstat(file).symlink? ? "#{file} -> #{File.readlink(file)}" : file
+end
+
+def generate_display_l_option(files)
+  files.map do |file|
+    [get_file_mode(file),
+     get_link_number(file).rjust(get_link_number_width(files) + 1),
+     get_user_name(file).ljust(get_user_name_width(files) + 1),
+     get_group_name(file).ljust(get_group_name_width(files) + 1),
+     get_byte(file).rjust(get_byte_width(files)),
+     get_last_modified_date(file),
+     get_file_name(file)].join(' ')
```

Comment:
このままでも良いと言えば良いですが、読みやすさと変更容易性を考えると以下のような書き方を推したいです。

```suggestion
    xxxx = [
      get_file_mode(file),
      get_link_number(file).rjust(get_link_number_width(files) + 1),
      get_user_name(file).ljust(get_user_name_width(files) + 1),
      get_group_name(file).ljust(get_group_name_width(files) + 1),
      get_byte(file).rjust(get_byte_width(files)),
      get_last_modified_date(file),
      get_file_name(file)
    ]
    xxxx.join(' ')
```